lib/engine/step/exchange.rb

```diff
@@ -42,17 +42,23 @@ def can_exchange?(entity, bundle = nil)
         owner = entity.owner
         return can_gain?(owner, bundle, exchange: true) if bundle
 
-        corporations =
-          ability.corporation == 'any' ? @game.corporations : [@game.corporation_by_id(ability.corporation)]
-
         shares = []
-        corporations.each do |corporation|
+        get_corporations(ability).each do |corporation|
```

Comment:
don't think you need this method, i think you can do

ability.corporation == 'any' ? @game.corporations : ability.corporations.map ...

Reply:
Rubocop was not pleased as it would not fit the tenary into one line (too long).

But instead I put this method into the Exchange ability class. And also added a closed? check which I think is needed. Please check the new attempt.